torch/serialization.py

```diff
@@ -865,19 +873,33 @@ def load(
             # If we want to actually tail call to torch.jit.load, we need to
             # reset back to the original position.
             orig_position = opened_file.tell()
+            overall_storage = None
             with _open_zipfile_reader(opened_file) as opened_zipfile:
                 if _is_torchscript_zip(opened_zipfile):
                     warnings.warn("'torch.load' received a zip file that looks like a TorchScript archive"
                                   " dispatching to 'torch.jit.load' (call 'torch.jit.load' directly to"
                                   " silence this warning)", UserWarning)
                     opened_file.seek(orig_position)
                     return torch.jit.load(opened_file, map_location=map_location)
+                if mmap:
+                    if not isinstance(f, str):
```

Comment:
f has type `Union[str, PathLike, BinaryIO, IO[bytes]]`

I think it makes sense to try to make this work for PathLike as well (and I will do this in a follow-up), but I don't think it makes sense for BinaryIO/IO[bytes]`, does that sound right?

Reply:
Sounds right to me.